### PR TITLE
Added idle protocol

### PR DESCRIPTION
**Pull request by anriha**
Added support for idle protocol. Also removed two events from output power management that weren't supposed to be there.

### CODE REVIEW

File: wlroots/ffi_build.py [hunk 1]
```diff
@@ -479,6 +479,71 @@ def has_xwayland() -> bool:
...
+    struct wl_event_loop *event_loop;
+    bool enabled;
+
+    struct wl_listener display_destroy;
+    struct {
+        struct wl_signal activity_notify;
+        struct wl_signal destroy;
+    } events;
+
+    void *data;
```
Comment (flacjacket) [inline wlroots/ffi_build.py:496]: Can you add `...;` to the end of this struct and the one below? That enables some extra consistency checking to be done when the cffi compile step happens.

**Review by flacjacket — CHANGES_REQUESTED** [review]
Looks good, just one minor tweak and we can merge this in!

File: wlroots/ffi_build.py (hunk 1 above)
Comment (anriha) [reply wlroots/ffi_build.py:496]: Done, I also added them to the output_power_management structs from my previous PR

**Review by flacjacket — APPROVED** [review]
Thanks!

**Pull request merged**